setup.py

```diff
@@ -2,204 +2,207 @@
-        except KeyError:
-            pass
-
+        dirs.extend(
+            Path(path)
+            for path in os.environ.get(envname, "").split(os.pathsep)
+            if path
+        )
 
     def add_from_flags(envname, flag_key, dirs):
-        for flag in os.environ.get(envname, "").split():
-            if flag.startswith(flag_key):
-                dirs.append(flag[len(flag_key):])
-
+        dirs.extend(
+            Path(flag[len(flag_key) :])
+            for flag in os.environ.get(envname, "").split()
+            if flag.startswith(flag_key)
+        )
 
-    if os.name == 'posix':
-        prefixes = ('/usr/local', '/sw', '/opt', '/opt/local', '/usr', '/')
+    if os.name == "posix":
+        prefixes = [
+            Path(path)
+            for path in (
+                "/usr/local",
+                "/sw",
+                "/opt",
+                "/opt/local",
+                "/usr",
+                "/",
+            )
+            if Path(path).is_dir()
+        ]
 
         default_header_dirs = []
         add_from_path("CPATH", default_header_dirs)
         add_from_path("C_INCLUDE_PATH", default_header_dirs)
         add_from_flags("CPPFLAGS", "-I", default_header_dirs)
         add_from_flags("CFLAGS", "-I", default_header_dirs)
-        default_header_dirs.extend(
-            os.path.join(_tree, 'include') for _tree in prefixes
-        )
+        default_header_dirs.extend(_tree / "include" for _tree in prefixes)
 
         default_library_dirs = []
         add_from_flags("LDFLAGS", "-L", default_library_dirs)
         default_library_dirs.extend(
-            os.path.join(_tree, _arch) for _tree in prefixes
-            for _arch in ('lib64', 'lib'))
+            _tree / _arch for _tree in prefixes for _arch in ("lib64", "lib")
+        )
         default_runtime_dirs = default_library_dirs
 
-    elif os.name == 'nt':
+    elif os.name == "nt":
         default_header_dirs = []  # no default, must be given explicitly
         default_library_dirs = []  # no default, must be given explicitly
         default_runtime_dirs = [  # look for DLL files in ``%PATH%``
-            _path for _path in os.environ['PATH'].split(';')]
+            Path(_path) for _path in os.environ["PATH"].split(";")
+        ]
         # Add the \Windows\system to the runtime list (necessary for Vista)
-        default_runtime_dirs.append('\\windows\\system')
+        # TODO: should that go to C:\ ?
+        default_runtime_dirs.append(Path("\\windows\\system"))
         # Add the \path_to_python\DLLs and tables package to the list
-        default_runtime_dirs.extend(
-            [os.path.join(sys.prefix, 'Lib\\site-packages\\tables')])
-
+        default_runtime_dirs.append(
+            Path(sys.prefix) / "Lib" / "site-packages" / "tables"
+        )
 
+    # TODO: still an issue?
     # Gcc 4.0.1 on Mac OS X 10.4 does not seem to include the default
     # header and library paths.  See ticket #18.
```

Comment:
Mac OS X 10.4 has been released in 2005, and clang is the default compiler since 2011 if I'm not wrong.
Anyway I can't say if that hack is still needed for all reasonably modern versions of OS X/Xcode.
I can only test Catalina if you want.

Reply:
Never used Mac, no idea. Just asking.